packages/react-aria-components/src/ComboBox.tsx

```diff
@@ -140,8 +140,8 @@ function ComboBoxInner<T extends object>({props, collection, comboBoxRef: ref}:
     listBoxRef,
     popoverRef,
     name: formValue === 'text' ? name : undefined
-  },
-  state);
+  }, state);
+  inputProps = filterDataAttrs(inputProps);
```

Comment:
Feels like it would be safer to perform this filtering on the props passed to the hook instead since the current approach relies on knowing that the hooks propagates the data attributes to `inputProps` specifically. I think I remember we discussed this in otters time, is there some details I'm forgetting from then that led us to this approach instead?

Reply:
my gut said allow all props to the hook but I think you have a good point about the reliance on knowing that this particular hook propagates

Reply:
If I'm understanding correctly, it feels like six of one, half a dozen on the other(which I think was the consensus in Otters). 

I made the change for this component to filter when passing to the hook. Pleas tell me if what I did was what you intended, and let me know if I should refactor the other changes to match filtering before the hook instead of after.